crates/uv-distribution/src/pyproject_mut.rs

```diff
@@ -34,47 +36,165 @@ impl PyProjectTomlMut {
     }
 
     /// Adds a dependency to `project.dependencies`.
-    pub fn add_dependency(&mut self, req: &Requirement) -> Result<(), Error> {
-        add_dependency(req, &mut self.doc["project"]["dependencies"])
+    pub fn add_dependency(
+        &mut self,
+        req: &Requirement,
+        source: Option<&Source>,
+    ) -> Result<(), Error> {
+        // Get or create `project.dependencies`.
+        let dependencies = self
```

Comment:
`toml_edit` does some implicit conversions when indexing directly on an `Item` which makes this form a lot better.. but very verbose. Maybe an extension trait could help for some of the common conversions.